File: Data_Structures/Maps_Hash_Dict/Amicable_Numbers.cpp
#include<iostream>
using namespace std;
int main(){                                                                                                          
int i,a,b,c=0;
cout<<"To check for the entered pair of numbers are Amicable Numbers or not.\n"<<endl;
cout<<"Enter any two Numbers : ";
cin>>a>>b;

for(i=1;i<=a/2;i++)
{
        if(a%i==0)
                c+=i;                                        
}
if(c==b)                                                                                                                      
        cout<<a<<" and "<<b<<" are Amicable Numbers."<<endl;
else
        cout<<a<<" and "<<b<<" are Not Amicable Numbers."<<endl;

cout<<"\n\nSome Amicable Numbers under 100,000 are :-"<<endl;
int x,y,j,m,n;
for(i=220;i<100000;i++){
        x=0,y=0;
        for(j=1;j<=i/2;j++){
                if(i%j==0){
                       x+=j;
                       //cout<<j<<endl;
                       }}
        m=1,n=0;
        while(m<=x/2){
                if(x%m==0)
                y+=m;
        m++;
        }
        if(i==y){
                if(x!=y){
                       cout<<x<<" and "<<y<<endl;
                       i+=800;    //to save time
        }}}
return 0;
}

File: Data_Structures/Maps_Hash_Dict/Amicable_Numbers.java

//Java program to check given numbers are Amicable Numbers or not

import java.util.*;

class AmicableNum{

	//Checks for valid Amicable Number
	boolean findAmicable(int num1, int num2){
		int sumofDiv1 =0;
		int sumofDiv2 =0;

		//Calculate sum of divisor for number 1
		for(int i=1; i <= num1/2; i++){

			if(num1%i==0){
				sumofDiv1 += i;
			}

		}

		//Calculate sum of divisor for number 2
		for(int j=1; j <= num2/2; j++){

			if(num2%j==0){
				sumofDiv2 += j;
			}

		}


		//Validate amicable numbers conditions
		if((num1 == sumofDiv2) && (num2 == sumofDiv1)){
			
			return true;

		}
		else{

			return false;

		}


	}

	public static void main(String[] args) {
		
		Scanner sc = new Scanner(System.in); //Create object

		int num1,num2;

		System.out.print("Enter first Number: ");
		num1 = sc.nextInt(); //first number

		System.out.print("Enter second Number: ");
		num2 = sc.nextInt(); //second number


		//Create object for class AmicableNum
		AmicableNum ami = new AmicableNum();

		//valdiating numbers
		if(ami.findAmicable(num1, num2)){

			System.out.println( num1 + " and " + num2 + " are Amicable Numbers.");

		}
		else{

			System.out.println( num1 + " and " + num2 + " are not Amicable Numbers.");

		}

	}
}